src/value/owned.rs

```diff
@@ -506,6 +506,68 @@ mod value_tests {
         assert_eq!(actual, expected);
     }
 
+    /// Each case is a set of tokens that are the same, and a set of tokens that are not ever equal to the first.
+    /// This should test symmetry/transitivity/commutativity
+    #[rstest(
+        // SymbolTokens with same text are equivalent
+        case::sym_text(
+            vec![
+                text_token("foo"),
+                text_token("foo").with_local_sid(10),
+                text_token("foo").with_local_sid(10).with_source("greetings", 2)
+            ],
+            vec![
+                text_token("bar"),
+                local_sid_token(10).with_source("greetings", 1),
+                local_sid_token(10).with_source("hello_table", 2)
+            ]
```

Comment:
I would put tokens without text here too, because all of them should be not equivalent. (e.g. local symbol with no text and local symbol with shared import and no text (with same source as above).

Reply:
I thought putting local symbol with no text i.e. `local_sid_token(200)` would be equal to `text_token("foo")` because for both `ImportSource` would be `None`. For local_symbol with a source I have already added `local_sid_token(10).with_source("greetings", 1)` for it. Does this make sense or am I missing something?

Reply:
It would be good to have a `local_sid_token` with no source or text for completeness